### PR TITLE
Sync/2023 03 03

### PR DESCRIPTION
**Pull request by BenceVirtonomy**
1. Typo fixes
2. "Include `acc_prior_ `in the time step calculation for shell and volume models" --> **we need to call the time step size calculation after all `acc_prior_  `updates, e.g. after external and contact force applications**
3. ShellContactDensity update
**4. Shell damping bug fix, they claim the time-stepping is stable for shell now**
5. `inner_relation.getSPHBody()` instead of `inner_relation.sph_body_`, same for `contact_relation.sph_body_` etc.
6. `upgradeToVector3D `renamed to `upgradeToVec3d`, upgradeToMatrix3D to upgradeToMat3d
8. Fabien, I kept our version of the contact density variable registration, will try to merge to upstream
9. Riemann solver update
10. Updates to constraint dynamics --> we keep using our internal version of those
11. **FluidViscousForceOnSolid renamed to ViscousForceFromFluid**, TotalForceOnSolid to TotalForceFromFluid, FluidViscousForceOnSolid to BaseForceFromFluid, etc.
12. **FluidForceOnSolidUpdate  to AllForceAccelerationFromFluid** 
13. No need to add the body part type as second template argument anymore for `SimpleDynamics`:

`SimpleDynamics<fluid_dynamics::EmitterInflowInjection> emitter_inflow_injection(emitter, 10, 0);` instead of 

`SimpleDynamics<fluid_dynamics::EmitterInflowInjection, BodyAlignedBoxByParticle> emitter_inflow_injection(emitter, 10, 0);`

14. ParticleData indices changed as follows:
image

### CODE REVIEW

File: SPHINXsys/src/shared/adaptations/adaptation.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
  * ------------------------------------------------------------------------*/
 /**
  * @file 	adaptation.h
- * @brief 	Adaptiation scheme for particle in multi-resolution scenario.
+ * @brief 	Adaptation scheme for particle in multi-resolution scenario.
```
Comment (YuVirtonomy) [inline SPHINXsys/src/shared/adaptations/adaptation.h:25]: maybe also change all adapt"i"ation to adaptation including the file name?

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/fluid_structure_interaction.h [hunk 2]
```diff
@@ -41,150 +41,129 @@ namespace SPH
...
 		};
 
 		/**
-		 * @class FluidViscousForceOnSolidInEuler
-		 * @brief Computing the viscous force from the fluid in eulerian framework
+		 * @class ViscousForceFromFluid
+		 * @brief Computing the viscous force from the fluid
 		 */
-		class FluidViscousForceOnSolidInEuler : public LocalDynamics, public FSIContactData
+		class ViscousForceFromFluid : public BaseForceFromFluid
```
Comment (YuVirtonomy) [inline SPHINXsys/src/shared/particle_dynamics/solid_dynamics/fluid_structure_interaction.h:66]: seem not suitable for multi-phase flow

File: SPHINXsys/src/shared/adaptations/adaptation.h (hunk 1 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/shared/adaptations/adaptation.h:25]: I think the file name is ok

File: SPHINXsys/src/shared/particle_dynamics/solid_dynamics/fluid_structure_interaction.h (hunk 2 above)
Comment (BenceVirtonomy) [reply SPHINXsys/src/shared/particle_dynamics/solid_dynamics/fluid_structure_interaction.h:66]: For this, please open an issue on prof. Hu's repo for him. Why not?

**Review by YuVirtonomy — APPROVED** [review]

**Pull request merged**